File: common/src/main/java/com/cyanspring/common/util/DailyKeyCounter.java
package com.cyanspring.common.util;

import java.util.Date;
import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.cyanspring.common.Clock;

public class DailyKeyCounter {
	private static final Logger log = LoggerFactory
			.getLogger(DailyKeyCounter.class);

	private int limit;
	private Date currentDate;
	private Map<String, Integer> countMap = new ConcurrentHashMap<String, Integer>();

	public DailyKeyCounter(int limit) {
		this.limit = limit;
		currentDate = getToday();
	}
	
	private Date getToday() {
		Date now = Clock.getInstance().now();
		return TimeUtil.getOnlyDate(now);
	}
	
	public boolean check(String key) {
		Date now = getToday();
		if(!currentDate.equals(now)) {
			countMap.clear();
			countMap.put(key, 1);
			currentDate = now;
			return true;
		}
		
		Integer count = countMap.get(key);
		if(null == count) {
			countMap.put(key, 1);
			return true;
		} 
		
		if(count >= limit)
			return false;
		
		countMap.put(key, count+1);
		return true;
	}
	
}

File: csharp/CSharpLTS/Common/Utils/DailyKeyCounter.cs
using System;
using System.Collections.Concurrent;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Common.Utils
{
    public class DailyKeyCounter
    {
        private int limit;
        private DateTime currentDate;
        private ConcurrentDictionary<string, int> countMap = new ConcurrentDictionary<string, int>();

        public DailyKeyCounter(int limit)
        {
            currentDate = getToday();
            this.limit = limit;
        }

        public bool check(string key)
        {
            DateTime now = getToday();
            if (DateTime.Compare(currentDate, now) != 0)
            {
                countMap.Clear();
                countMap.TryAdd(key, 1);
                currentDate = now;
                return true;
            }
            if (!countMap.ContainsKey(key))
            {
                countMap.TryAdd(key, 1);
                return true;
            }
            int count = countMap[key];
            if (count >= limit)
            {
                return false;
            }
            countMap[key] = count + 1;
            return true;
        }

        private DateTime getToday()
        {
            return DateTime.Today;
        }

    }
}
